src/DDTrace/Integrations/AMQP/AMQPIntegration.php

```diff
@@ -167,6 +167,63 @@ function ($This) use ($integration) {
             ]
         );
 
+        trace_method(
+            "PhpAmqpLib\Channel\AMQPChannel",
+            "batch_basic_publish",
+            [
+                'prehook' => function (SpanData $span, $args) use ($integration) {
+                    /** @var AMQPMessage $message */
+                    $message = $args[0];
+                    if (!is_null($message)) {
+                        $integration->injectContext($message);
+                    }
+                },
+                'posthook' => function (SpanData $span, $args, $exception) use ($integration) {
+                    /** @var AMQPMessage $message */
+                    $message = $args[0];
+                    /** @var string $exchange */
+                    $exchange = $args[1];
+                    /** @var string $routing_key */
+                    $routingKey = $args[2] ?? '';
+
+                    $exchangeDisplayName = $integration->formatExchangeName($exchange);
+                    $routingKeyDisplayName = $integration->formatRoutingKey($routingKey);
+
+                    $integration->setGenericTags(
+                        $span,
+                        'batch_basic.add',
```

Comment:
Considering that the span created from `basic_publish` is `basic.publish`, I believe that the operation name for this span should be `batch_basic.publish`.

Was there any reason to use `.add` instead of `.publish`?

Reply:
I wanted to differentiate the actions that accumulates data in memory from the ones that actually interact with the server - IMHO the naming in the underlying library is misleading as `batch_basic_publish()` doesn't publish anything until `publish_batch()` is called.

Let me know what's your preference and I can adjust it.

Reply:
Makes sense :+1: That's good for me then :)